### ISSUE TITLE
[Security] XSS attack through user renaming

### ISSUE THREAD

**Issue by Mickael-van-der-Beek** [seripap/darkwire.io#47]
In Darkwire there is a feature where a user can rename himself from the originally assigned random alias to any username he'd like provided it is smaller than 16 characters:

https://github.com/seripap/darkwire.io/blob/master/src/room.js#L90

On the other hand, this length check is not done when the user initially connects to the room which is an issue because the username is generated by the client and not the server:

https://github.com/seripap/darkwire.io/blob/master/src/room.js#L40

Proof of concept:

```
const io = require('socket.io-client');
const socket = io.connect('http://127.0.0.1:3000/$ROOM_ID');

socket.emit('add user', {
  username: '<script>alert(1)</script>',
  publicKey: ''
});

setTimeout(() => socket.disconnect(), 3000);
```

Just replace the "$ROOM_ID" placeholder with the room id of a Darkwire room you are currently connected to in your browser and run the script.

Although this attack requires the room id, it still compromises any end user that opens a chat with the attacker. From the other security vulnerability issue #46 I opened we can see that it is possible for the app to leak the room id information so relying to much on this single line of defense might not be enough. 

To fix this issue, the server has to validate that usernames are only alphanumeric and in a certain length range.

**Comment by seripap** [comment]
Thanks for the report. I will fix this soon.

**Comment by seripap** [comment]
Fixed in [v1.5.8](https://github.com/seripap/darkwire.io/releases/tag/v1.5.8)
